src/main/java/fr/sncf/osrd/speedcontroller/generators/ConstructionAllowanceGenerator.java

```diff
@@ -51,7 +52,9 @@ protected double getFirstGuess() {
         // lowering it twice compared to its approximate value
         // keeping getFirstLowEstimate() at 0.0 prevents mistakes
         var time = evalRunTime(sim, schedule, maxSpeedControllers);
-        return ((this.getFirstHighEstimate() - this.getFirstLowEstimate()) * time / (time + 2 * value));
+        if (value != 0.0)
```

Comment:
True, you're right. I actually wrote that because of a margin test of 0 that made the test crash, but actually this won't happen when using the feature.